packages/ozone/src/team/index.ts

```diff
@@ -0,0 +1,208 @@
+import Database from '../db'
+import { Selectable } from 'kysely'
+import { Member } from '../db/schema/member'
+import { Member as TeamMember } from '../lexicon/types/tools/ozone/team/defs'
+import { ProfileViewDetailed } from '../lexicon/types/app/bsky/actor/defs'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+import { chunkArray } from '@atproto/common'
+import AppContext from '../context'
+import { httpLogger } from '../logger'
+
+export type TeamServiceCreator = (db: Database) => TeamService
+
+export class TeamService {
+  constructor(public db: Database) {}
+
+  static creator() {
+    return (db: Database) => new TeamService(db)
+  }
+
+  async list({
+    cursor,
+    limit = 25,
+  }: {
+    cursor?: string
+    limit?: number
+  }): Promise<{ members: Selectable<Member>[]; cursor?: string }> {
+    let builder = this.db.db.selectFrom('member').selectAll()
+    if (cursor) {
+      builder = builder.where('createdAt', '>', new Date(cursor))
+    }
+    const members = await builder
+      .limit(limit)
+      .orderBy('createdAt', 'asc')
+      .execute()
+
+    return { members, cursor: members.slice(-1)[0]?.createdAt.toISOString() }
+  }
+
+  async create({
+    role,
+    did,
+    disabled,
+    updatedAt,
+    createdAt,
+    lastUpdatedBy,
+  }: Omit<Selectable<Member>, 'createdAt' | 'updatedAt'> & {
+    createdAt?: Date
+    updatedAt?: Date
+  }): Promise<Selectable<Member>> {
+    const now = new Date()
+    const newMember = await this.db.db
+      .insertInto('member')
+      .values({
+        role,
+        did,
+        disabled,
+        lastUpdatedBy,
+        updatedAt: updatedAt || now,
+        createdAt: createdAt || now,
+      })
+      .returningAll()
+      .executeTakeFirstOrThrow()
+
+    return newMember
+  }
+
+  async upsert({
+    role,
+    did,
+    lastUpdatedBy,
+  }: Pick<
+    Selectable<Member>,
+    'role' | 'did' | 'lastUpdatedBy'
+  >): Promise<void> {
+    const now = new Date()
+    await this.db.db
+      .insertInto('member')
+      .values({
+        role,
+        did,
+        lastUpdatedBy,
+        disabled: false,
+        updatedAt: now,
+        createdAt: now,
+      })
+      .onConflict((oc) => oc.column('did').doUpdateSet({ role }))
+      .execute()
+  }
+
+  async update(
+    did: string,
+    updates: Partial<
+      Pick<
+        Selectable<Member>,
+        'role' | 'disabled' | 'lastUpdatedBy' | 'updatedAt'
+      >
+    >,
+  ): Promise<Selectable<Member>> {
+    const updatedMember = await this.db.db
+      .updateTable('member')
+      .where('did', '=', did)
+      .set({
+        ...updates,
+        updatedAt: updates.updatedAt || new Date(),
+      })
```

Comment:
There is a bit of a security issue here, since `updates` may contain fields not explicitly `Pick`ed out of `Selectable<Member>`, and they end up passed directly through to the query.  For example, imagine if `did` were a field on `updates`.

Looking at the `team.updateMember` endpoint, I think the user could end-up successfully changing the user's `createdAt` time through the API.  On its own it that isn't a critical issue, but as we add new columns to the `members` table we might run into more sensitive issues.